godtossing/Pos/Ssc/GodTossing/Toss/Base.hs

```diff
@@ -468,10 +473,22 @@ checkCertificatesPayload
 checkCertificatesPayload epoch certs = do
     richmenSet <- getKeys <$> (note (NoRichmen epoch) =<< getRichmen epoch)
     exceptGuardM CertificateAlreadySent
-        (notM hasCertificateToss) (HM.keys certs)
+        (notM hasCertificateToss) (HM.keys (getVssCertificatesMap certs))
     exceptGuardSnd CertificateNotRichmen
         ((`HS.member` richmenSet) . addressHash . vcSigningKey)
-        (HM.toList certs)
+        (HM.toList (getVssCertificatesMap certs))
+    existingVssKeys <-
+        HS.fromList . map vcVssKey . toList . getVssCertificatesMap <$>
+        getVssCertificates
+    exceptGuardSnd CertificateDuplicateVssKey
+        (not . (`HS.member` existingVssKeys) . vcVssKey)
+        (HM.toList (getVssCertificatesMap certs))
+    let newVssKeys =
+            HS.fromList . map vcVssKey . toList . getVssCertificatesMap $
+            certs
+    exceptGuardSnd CertificateDuplicateVssKey
+        (not . (`HS.member` newVssKeys) . vcVssKey)
+        (HM.toList (getVssCertificatesMap certs))
```

Comment:
This check seems to always fail.
Also it's probably useless, because for block processing it's ensured by deserialization and for data processing it's ensured by the check above.
But I guess it shouldn't be expensive at all, so I don't mind having it here (but it should be fixed, I am 99% sure).

Reply:
as said Vanya it seems to always fail, moreover I don't see a necessity to check this statement for each map separately, b/c AFAIU map can't contain duplicated elements, however, they can contain duplicates in union